activation/atxdb.go

```diff
@@ -94,6 +96,50 @@ func (m *ActivationDb) CalcActiveSetFromView(a *types.ActivationTx) (uint32, err
 
 }
 
+const NIPST_LAYERTIME = 1000
+
+func (db *ActivationDb) ValidateAtx(atx *types.ActivationTx) error {
+	// validation rules: no other atx with same id and sequence number
+	// if s != 0 the prevAtx is valid and it's seq num is s -1
+	// positioning atx is valid
+	// validate nipst duration?
+	// fields 1-7 of the atx are the challenge of the poet
+	// layer index i^ satisfies i -i^ < (layers_passed during nipst creation) ANTON: maybe should be ==?
+	// the atx view contains d active Ids
+
+	eatx, _ := db.GetAtx(atx.Id())
+	if eatx != nil {
+		return fmt.Errorf("Found atx with same id")
+	}
+	prevAtxIds, err := db.GetNodeAtxIds(atx.NodeId)
+	if err == nil && len(prevAtxIds) > 0 {
+		prevAtx, err := db.GetAtx(prevAtxIds[len(prevAtxIds)])
```

Comment:
`prevAtxIds[len(prevAtxIds)]` will always panic.

If you have a 3 element slice (`len()==3`) it has elements `0`, `1` and `2`, but never `3` (or its length would be 4).